protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/encoding/MsgPackConverter.java

```diff
@@ -77,7 +77,11 @@ public static byte[] convertToMsgPack(final InputStream inputStream) {
 
       return outputStream.toByteArray();
     } catch (final Exception e) {
-      throw new RuntimeException("Failed to convert JSON to MessagePack", e);
+      if (e instanceof IllegalArgumentException) {
+        throw new IllegalArgumentException("Failed to convert JSON to MessagePack", e);
+      } else {
+        throw new RuntimeException("Failed to convert JSON to MessagePack", e);
+      }
```

Comment:
❓ I'm a bit confused about the changes, why is this necessary? Is it only to change it to an `IllegalArgumentException` so we have a specific exception we can catch and not change the behavior of other `RuntimeExceptions`?

Reply:
https://github.com/camunda/zeebe/issues/9832 got `IllegalArgumentException` exception, I'm not sure the others may get `RuntimeExceptions` exception  or not, so try specific `IllegalArgumentException ` exception to catch.